napari/layers/_image_layer/model.py

```diff
@@ -94,6 +94,8 @@ class Image(Layer):
 
     _colormaps = AVAILABLE_COLORMAPS
 
+    class_keymap = {}
```

Comment:
Can this variable not be provided by the Mixin directly? It's a bit noisy that it has to be added to any subclass in the hierarchy...

Reply:
Dictionaries are mutable so it would create quite the mess if we did that. We could resolve this by using a custom metaclass but I'm not so keen on that either.

Reply:
I was thinking something like overriding `__new__` so that if the class has a dictionary, use that, but if it doesn't, add one? Is that possible? I'm not too clear on class variables in Python actually!

Reply:
That's a clever way to do it but it unfortunately won't work as expected when applied to subclasses; e.g. `Pyramid` would share the same bindings as `Image`. I think we could additionally do something hacky and iterate through a class's bases, checking if one of their `class_keybindings` is the same as its own via the `is` operator. Would you like me to try that?